src/js.ts

```diff
@@ -15,12 +15,13 @@ type status =
   | "comment"
   | "line-comment";
 
-type Visitor = (type: breakpoints, index: number) => false | void;
-
-export default function analyze(source: string, visitor: Visitor) {
+export function* topLevel(
```

Comment:
I'm not sure about the named export. `analyze` is not a good name, but `topLevel` is not descriptive enough. And due it's a module with a single function, probably should be exported as `default`.

Reply:
Oh, I can change it back to a default export - just usually go for named ones myself and didn't catch it here.

I agree about the name - what would you suggest here? Perhaps `iterateTopLevel()`?